src/components/data/SaveAsField.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * @author sriram
+ *
+ * A field to collect path (with filename) from a user to save a newly created file.
+ */
+import React from "react";
+
+import { useTranslation } from "i18n";
+
+import ids from "./ids";
+import styles from "./styles";
+import withErrorAnnouncer from "components/utils/error/withErrorAnnouncer";
+
+import {
+    build as buildDebugId,
+    FormTextField,
+    getFormError,
+} from "@cyverse-de/ui-lib";
+
+import {
+    Button,
+    Grid,
+    IconButton,
+    InputAdornment,
+    makeStyles,
+} from "@material-ui/core";
+
+import ClearIcon from "@material-ui/icons/Clear";
+import SaveAsDialog from "./SaveAsDialog";
+
+const useStyles = makeStyles(styles);
+
+const SaveAsButton = (props) => {
+    const { baseId, startingPath, onConfirm } = props;
+    const { t } = useTranslation("data");
+    const [open, setOpen] = React.useState(false);
+
+    return (
+        <>
+            <Button
+                id={buildDebugId(baseId, ids.SAVE_AS_BTN)}
+                color="primary"
+                size="small"
+                variant="outlined"
+                onClick={() => setOpen(true)}
+            >
+                {t("saveas")}
+            </Button>
+            <SaveAsDialog
+                path={startingPath}
+                open={open}
+                onClose={() => setOpen(false)}
+                onSaveAs={(path) => {
+                    onConfirm(path);
+                    setOpen(false);
+                }}
+            />
+        </>
+    );
+};
+
+const SaveAsField = ({ startingPath, showErrorAnnouncer, ...props }) => {
+    // These props need to be spread down into the FormTextField
+    const {
+        id,
+        field: { name, value },
+        form: { touched, errors, setFieldValue },
+        required,
+    } = props;
+    const classes = useStyles();
+    const { t } = useTranslation("data");
+    const errorMsg = getFormError(name, touched, errors);
+
+    const inputProps = {
+        readOnly: true,
+        endAdornment: (
+            <InputAdornment position="end">
+                <SaveAsButton
+                    baseId={id}
+                    startingPath={startingPath}
+                    multiSelect={false}
+                    name={name}
+                    onConfirm={(path) => {
+                        setFieldValue(name, path);
+                    }}
+                />
+            </InputAdornment>
+        ),
+    };
```

Comment:
While this is a look alike component of `InputSelector`, I wouldn't go too far into having a single component for Input selection and Save as field as they are fundamentally for different purpose( InputSelector collects an existing path while SaveAsField get a new file name from the user). I don't anticipate this buying us much at this much other than saving few lines of code trading for few hours of work.

Reply:
👌 I agree the buttons and the dialogs they open are fundamentally different, though the form field that wraps those buttons are exactly the same. In a previous comment you mentioned refactoring the `SaveAsField`, so I thought the `InputSelector` could be refactored at the same time, but it's a minor improvement, so I'm fine with skipping that for now.

I think at least the Grids wrapping both form fields could be removed now, since they look like unneeded cruft at this point.